packages/core/src/2d/text/TextRenderer.ts

```diff
@@ -0,0 +1,531 @@
+import { BoundingBox, Color, Vector2, Vector3 } from "@oasis-engine/math";
+import { Sprite, SpriteMaskInteraction, SpriteMaskLayer } from "..";
+import { CompareFunction, Renderer, Shader, UpdateFlag } from "../..";
+import { Camera } from "../../Camera";
+import { assignmentClone, deepClone, ignoreClone } from "../../clone/CloneManager";
+import { Entity } from "../../Entity";
+import { ShaderProperty } from "../../shader/ShaderProperty";
+import { Texture2D } from "../../texture";
+import { TextHorizontalAlignment, TextVerticalAlignment } from "../enums/TextAlignment";
+import { TextHorizontalOverflow, TextVerticalOverflow } from "../enums/TextOverflow";
+import { TextUtils } from "./TextUtils";
+
+export class TextRenderer extends Renderer {
+  private static _textureProperty: ShaderProperty = Shader.getPropertyByName("u_spriteTexture");
+  private static _tempVec2: Vector2 = new Vector2();
+  private static _tempVec3: Vector3 = new Vector3();
+
+  /** @internal temp solution. */
+  @ignoreClone
+  _customLocalBounds: BoundingBox = null;
+  /** @internal temp solution. */
+  @ignoreClone
+  _customRootEntity: Entity = null;
+
+  @assignmentClone
+  private _sprite: Sprite = null;
+  @deepClone
+  private _positions: Vector3[] = [new Vector3(), new Vector3(), new Vector3(), new Vector3()];
+  @deepClone
+  private _color: Color = new Color(1, 1, 1, 1);
+  @assignmentClone
+  private _text: string = "";
+  @assignmentClone
+  private _width: number = 0;
+  @assignmentClone
+  private _height: number = 0;
+  @assignmentClone
+  private _fontName: string = "Arial";
+  @assignmentClone
+  private _fontSize: number = 24;
+  @assignmentClone
+  private _lineSpace: number = 0;
+  @assignmentClone
+  private _isBold: boolean = false;
+  @assignmentClone
+  private _isItalic: boolean = false;
+  @assignmentClone
+  private _horizontalAlignment: TextHorizontalAlignment = TextHorizontalAlignment.Center;
+  @assignmentClone
+  private _verticalAlignment: TextVerticalAlignment = TextVerticalAlignment.Center;
+  @assignmentClone
+  private _horizontalOverflow: TextHorizontalOverflow = TextHorizontalOverflow.Overflow;
+  @assignmentClone
+  private _verticalOverflow: TextVerticalOverflow = TextVerticalOverflow.Overflow;
+  @ignoreClone
+  private _dirtyFlag: number = DirtyFlag.Style;
+  @ignoreClone
+  private _isWorldMatrixDirty: UpdateFlag;
+  @assignmentClone
+  private _maskInteraction: SpriteMaskInteraction = SpriteMaskInteraction.None;
+  @assignmentClone
+  private _maskLayer: number = SpriteMaskLayer.Layer0;
+
+  /**
+   * Rendering color for the TextRenderer.
```

Comment:
Rendering color for the Text.  Not TextRenderer.

Reply:
Done